dsindex.go

```diff
@@ -88,3 +90,38 @@ func LoadIndex(data []byte) (DatastoreIndex, error) {
 
 	return index, nil
 }
+
+func LoadIndexDatastore(resp dsadmin.ListIndexesResponse, readyOnly bool) (DatastoreIndex, error) {
+	index := make(DatastoreIndex, len(resp.Indexes))
+	for _, spec := range resp.Indexes {
+		if !readyOnly || spec.State == dsadmin.Index_READY {
+			entIndex := entityIndex{ancestor: spec.Ancestor == 2, fields: make(map[string]fieldIndex, len(spec.Properties))}
```

Comment:
Should we make 
```suggestion
			entIndex := entityIndex{ancestor: spec.Ancestor == dsadmin.Index_ALL_ANCESTORS, fields: make(map[string]fieldIndex, len(spec.Properties))}
```
?

Reply:
Ah yep good call outs